contracts/modules/cover/Cover.sol

```diff
@@ -153,19 +165,32 @@ contract Cover is ICover, MasterAwareV2, IStakingPoolBeacon, ReentrancyGuard, Mu
       allocationRequest.globalMinPrice = GLOBAL_MIN_PRICE_RATIO;
     }
 
+    uint nxmPriceInCoverAsset = pool().getTokenPriceInAsset(params.coverAsset);
+
     uint previousSegmentAmount;
 
+    uint coverAmountInCoverAsset;
+    uint amountDueInNXM;
+
     if (params.coverId == 0) {
 
-      // new cover
+      // Handle creating a new cover
       coverId = coverNFT.mint(params.owner);
       _coverData[coverId] = CoverData(params.productId, params.coverAsset, 0 /* amountPaidOut */);
 
+      allocationRequest.coverId = coverId;
+      // for a new cover, the first segment the remaining period and new period are the requested period
+      allocationRequest.remainingPeriod = uint32(allocationRequest.period);
+      allocationRequest.newPeriod = uint32(allocationRequest.period);
```

Comment:
these should't be used at all for new covers, do we really need to set them?

Reply:
Good question - The way i wrote requestAllocation is that it handles both create and edit with the same code paths by using

allocationRequest.remainingPeriod  as the period used to compute premium for the New Cover case.

Seeing that this may cause confusion - i'm thinking to basically have an if/else in request allocation and have 2 totally separate code paths for computing price for new cover and edit cover and not use these fields in both cases